wallet/benchmark_helpers_test.go

```diff
@@ -44,102 +72,23 @@ func exponentialGrowth(i int) int {
-	// would helpful in determining the dynamic padding for the UTXO digits.
-	maxUTXOs int
-
-	// maxAddresses is the maximum number of addresses in the benchmark
-	// series. That would helpful in determining the dynamic padding for the
-	// address digits.
-	maxAddresses int
-}
-
-// name returns a dynamically generated benchmark name based on accounts,
-// UTXOs, and addresses. Uses dynamic padding based on maximum values for
-// proper sorting in visualization tools.
-func (b benchmarkDataSize) name(namingInfo benchmarkNamingInfo) string {
-	accountDigits := len(strconv.Itoa(namingInfo.maxAccounts))
-
-	name := fmt.Sprintf("%0*d-Accounts", accountDigits, b.numAccounts)
-
-	if b.numAddresses > 0 {
-		addressDigits := len(strconv.Itoa(namingInfo.maxAddresses))
-		name += fmt.Sprintf("-%0*d-Addresses", addressDigits,
-			b.numAddresses)
+// mapRange maps fn over indices [start..end] (inclusive) and returns the
+// results. This provides functional-style array generation for benchmarks.
+//
+//nolint:unparam
+func mapRange(start, end int, fn growthFunc) []int {
+	result := make([]int, end-start+1)
+	for i := range result {
+		result[i] = fn(start + i)
 	}
 
-	if b.numUTXOs > 0 {
-		utxoDigits := len(strconv.Itoa(namingInfo.maxUTXOs))
-		name += fmt.Sprintf("-%0*d-UTXOs", utxoDigits, b.numUTXOs)
-	}
-
-	return name
+	return result
 }
 
-// benchmarkConfig holds configuration for benchmark wallet setup.
-type benchmarkConfig struct {
-	// accountGrowth is the function to use to grow the number of accounts.
-	accountGrowth growthFunc
-
-	// utxoGrowth is the function to use to grow the number of UTXOs.
-	utxoGrowth growthFunc
-
-	// addressGrowth is the function to use to grow the number of addresses.
-	addressGrowth growthFunc
-
-	// maxIterations is the maximum number of iterations to run.
-	maxIterations int
-
-	// startIndex is the index to start the benchmark at.
-	startIndex int
-}
-
-// generateBenchmarkSizes creates benchmark data sizes programmatically.
-func generateBenchmarkSizes(
-	config benchmarkConfig) ([]benchmarkDataSize, benchmarkNamingInfo) {
-
-	var sizes []benchmarkDataSize
-
-	// Calculate maximum values for proper padding.
-	maxAccounts := config.accountGrowth(config.maxIterations)
-	maxUTXOs := config.utxoGrowth(config.maxIterations)
-	maxAddresses := config.addressGrowth(config.maxIterations)
-
-	namingInfo := benchmarkNamingInfo{
-		maxAccounts:  maxAccounts,
-		maxUTXOs:     maxUTXOs,
-		maxAddresses: maxAddresses,
-	}
-
-	for i := config.startIndex; i <= config.maxIterations; i++ {
-		sizes = append(sizes, benchmarkDataSize{
-			numAccounts:  config.accountGrowth(i),
-			numUTXOs:     config.utxoGrowth(i),
-			numAddresses: config.addressGrowth(i),
-		})
-	}
-
-	return sizes, namingInfo
+// calculatePadding returns the number of characters needed to pad values
+// up to the given maximum value.
+func calculatePadding(maxValue int) int {
+	return len(strconv.Itoa(maxValue))
 }
```

Comment:
I think we could rename this function to make it more concise and aligned with what it actually does. From my perspective, it returns the number of characters required to write the string, not the amount of padding needed.

```suggestion
// decimalWidth returns the number of characters in the decimal
// representation of `value`
func decimalWidth(value int) int {
	return len(strconv.Itoa(value))
}
```